packages/sdk/schema/src/testing/generator.ts

```diff
@@ -149,27 +147,27 @@ export const createObjectPipeline = <T extends BaseObject>(
 ): ((obj: ExcludeId<T>) => Effect.Effect<Live<T>, never, never>) => {
   if (!db) {
     return (obj: ExcludeId<T>) => {
-      const pipeline: Effect.Effect<Live<T>> = pipe(
-        Effect.succeed(obj),
-        // Effect.tap(logObject('before')),
-        Effect.map((obj) => createProps(generator, type, optional)(obj)),
-        Effect.map((obj) => createReactiveObject(type)(obj)),
-        // Effect.tap(logObject('after')),
-      );
+      const pipeline: Effect.Effect<Live<T>> = Effect.gen(function* (_) {
+        // logObject('before')(obj);
+        const withProps = createProps(generator, type, optional)(obj);
+        const liveObj = createReactiveObject(type)(withProps);
+        // logObject('after')(liveObj);
+        return liveObj;
+      });
 
       return pipeline;
     };
   } else {
     return (obj: ExcludeId<T>) => {
-      const pipeline: Effect.Effect<ReactiveEchoObject<any>, never, never> = pipe(
-        Effect.succeed(obj),
-        // Effect.tap(logObject('before')),
-        Effect.map((obj) => createProps(generator, type, optional)(obj)),
-        Effect.map((obj) => createReactiveObject(type)(obj)),
-        Effect.flatMap((obj) => Effect.promise(() => createReferences(type, db)(obj))),
-        Effect.map((obj) => addToDatabase(db)(obj)),
-        // Effect.tap(logObject('after')),
-      );
+      const pipeline: Effect.Effect<ReactiveEchoObject<any>, never, never> = Effect.gen(function* (_) {
+        // logObject('before')(obj);
+        const withProps = createProps(generator, type, optional)(obj);
+        const liveObj = createReactiveObject(type)(withProps);
+        const withRefs = yield* Effect.promise(() => createReferences(type, db)(liveObj));
+        const dbObj = addToDatabase(db)(withRefs);
+        // logObject('after')(dbObj);
+        return dbObj;
+      });
```

Comment:
what's the change here? did this just change so you could console log and could be reverted? I suspect they must have a logger that could be used in a pipeline

Reply:
I switched to the generator syntax, because the `Effect.` `map` / `flatMap` aspect is just noise.

Reply:
ah I see now, github's diff highlighting is brutal here

Reply:
Yeah it looks way more gnarly than it is.